cpp/memilio/io/result_io.h

```diff
@@ -86,11 +91,121 @@ class SimulationResult
 };
 
 /**
- * @brief read simulation result from h5 file
- * @param filename name of file
+ * @brief Read simulation result from h5 file.
+ * @param filename name of the file to be read.
  */
 IOResult<std::vector<SimulationResult>> read_result(const std::string& filename);
 
+/**
+ * @brief Save the results of a parameter study.
+ * Stores different percentiles (p5, p25, p50, p75, p90) and sums of the results and parameters. 
+ * @param ensemble_results Result of each simulation run.
+ * @param ensemble_params Üarameters used for each simulation run.
+ * @param county_ids Ids of the county nodes.
+ * @param result_dir Top level directory for all results of the parameter study.
+ * @param save_single_runs [Default: true] Defines if single run results are written to the disk.
+ * @param save_single_runs [Default: true] Defines if percentiles are written to the disk.
+ * @return Any io errors that occur during writing of the files.
+ */
+template <class Model>
+IOResult<void> save_results(const std::vector<std::vector<TimeSeries<double>>>& ensemble_results,
+                            const std::vector<std::vector<Model>>& ensemble_params,
+                            const std::vector<int>& county_ids, const fs::path& result_dir,
+                            bool save_single_runs = true, bool save_percentiles = true)
+{
+    //save results and sum of results over nodes
+    auto ensemble_result_sum = sum_nodes(ensemble_results);
+    auto num_groups          = (int)(size_t)ensemble_params[0][0].parameters.get_num_groups();
+    if(save_single_runs)
+    {
+        for (size_t i = 0; i < ensemble_result_sum.size(); ++i) {
+        BOOST_OUTCOME_TRY(save_result(ensemble_result_sum[i], {0}, num_groups,
+                                           (result_dir / ("results_run" + std::to_string(i) + "_sum.h5")).string()));
+        BOOST_OUTCOME_TRY(save_result(ensemble_results[i], county_ids, num_groups,
+                                           (result_dir / ("results_run" + std::to_string(i) + ".h5")).string()));
+        }
```

Comment:
@HenrZu Let's have a look on this on Monday. Isn't this actually already saved in the run function?